frontend/app/routes/_gcweb-app.apply.date-of-birth.tsx

```diff
@@ -0,0 +1,122 @@
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-1122',
+  pageTitleI18nKey: 'intake-forms:date-of-birth.page-title',
+} as const satisfies RouteHandleData;
+export async function loader({ request }: LoaderFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  return json({ ok: true });
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  const dateOfBirthFormSchema = z.object({
+    year: z
+      .string()
+      .min(1, { message: 'empty-field' })
+      .transform((val) => val.trim()),
+    month: z
+      .string()
+      .min(1, { message: 'empty-field' })
+      .transform((val) => val.trim()),
+    day: z
+      .string()
+      .min(1, { message: 'empty-field' })
+      .transform((val) => val.trim()),
+  });
+  const formData = Object.fromEntries(await request.formData());
+
+  const dateOfBirth = formData['year'] + '-' + formData['month'] + '-' + formData['day']; //2024-11-14
+  const dateSchema = z.coerce.date();
+  const dateParsedResult = dateSchema.safeParse(dateOfBirth);
+
+  if (!dateParsedResult.success) {
+    return json({
+      errors: 'invalid-date',
+      formData: formData as Partial<z.infer<typeof dateOfBirthFormSchema>>,
+    });
+  }
+
+  //TODO
+  //COMPLETE ONCE THE NEXT PAGE IS ADDED
+
+  return redirect('/apply/date-of-birth', {});
+}
+
+export default function EnterDoB() {
+  const actionData = useActionData<typeof action>();
+  const errorSummaryId = 'error-summary';
+  const { t } = useTranslation(i18nNamespaces);
+  /**
+   * Gets an error message based on the provided internationalization (i18n) key.
+   *
+   * @param errorI18nKey - The i18n key for the error message.
+   * @returns The corresponding error message, or undefined if no key is provided.
+   */
+  function getErrorMessage(errorI18nKey?: string): string | undefined {
+    if (!errorI18nKey) return undefined;
+
+    /**
+     * The 'as any' is employed to circumvent typechecking, as the type of
+     * 'errorI18nKey' is a string, and the string literal cannot undergo validation.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    return t(`intake-forms:date-of-birth.error-message.${errorI18nKey}` as any);
+  }
+  const errorMessages = {
+    dateOfBirthFieldSet: getErrorMessage(actionData?.errors),
+  };
+  const errorSummaryItems = createErrorSummaryItems(errorMessages);
+
+  useEffect(() => {
+    if (actionData?.formData && hasErrors(actionData.formData)) {
+      scrollAndFocusToErrorSummary(errorSummaryId);
+    }
+  }, [actionData]);
+
+  return (
+    <>
+      {errorSummaryItems.length > 0 && <ErrorSummary id={errorSummaryId} errors={errorSummaryItems} />}
+      <Form method="post">
+        <div>
+          <p>{t('intake-forms:date-of-birth.enter-dob-message')}</p>
+          <fieldset id="dateOfBirthFieldSet">
```

Comment:
```suggestion
          <fieldset id="dateOfBirthFieldSet">
            <legend>{t('intake-forms:date-of-birth.enter-dob-message')}</legend>
```

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/legend

Reply:
done